include/deal.II/grid/reference_cell.h

```diff
@@ -183,6 +183,77 @@ namespace ReferenceCell
     return Point<dim>(+0.0, +0.0, +0.0);
   }
 
+  /**
+   * Return i-th unit tangential vector (array of dim-1 elements of Point<dim>)
+   * of a face of the reference cell. The vectors are arranged such that the
+   * cross product between the two vectors return the unit normal vector.
+   */
+  template <int dim>
+  inline Tensor<1, dim>
+  unit_tangential_vectors(const Type &       reference_cell,
+                          const unsigned int face_no,
+                          const unsigned int i)
+  {
+    if (reference_cell == get_hypercube(dim))
```

Comment:
Can you add a check that the given `dim` argument matches the dimensionality of `reference_cell`?

Reply:
This line is exactly doing this implicitly. The function `get_hypercube()` return the reference-cell type of the right dimension so that if that does not matches the comparison will fail.

Reply:
What I meant is: Add such an assertion before the first `if`. Right now, if you call this function with `dim==2` and `reference_cell==tet`, you'll run into the `ExcNotImplemented()` at the very bottom, but that's not helpful to the user: it isn't that the necessary code isn't implemented, but that the function just doesn't make sense with this combination of arguments.